### ISSUE TITLE
codeql fails with exit code 32 for c language analysis under the macOS environment

### ISSUE THREAD

**Issue by xwm1992** [github/codeql#14703]
**Description of the issue**

Hello codeql community, 

our [community project](https://github.com/apache/eventmesh) has merged the c language code to the repository, but after that each pull request faced the ci check error, 
<img width="1672" alt="image" src="https://github.com/github/codeql/assets/13237619/e9f74c1c-e64c-4ebd-a396-1f8e3d13d47c">

and we try to fix this issue at the pull request[1] below, but it still can't pass the codeql perform at the macOS environment [2], please help take a look at this issue, maybe we ignore something, but the UbuntuOS has passed the ci check, make us confused, please help to solve this issue, thanks a lot.

[1]https://github.com/apache/eventmesh/pull/4543
[2]https://github.com/apache/eventmesh/actions/runs/6768882396/job/18394170320

**Comment by aibaars** [comment]
The problem is that the `matrix` does not define a `language` property, so `maxtrix.language` is `null`. Therefore the `Build C` step is skipped and CodeQL can't analyze the C code. 

I also see you defined  `languages: ${{ matrix.language }}` in the `codeql-action/init` step. Because `matrix.language` is not defined, this causes the `codeql-action/init` step to enable the analysis of all languages found in the repository.

**Comment by aibaars** [comment]
> could you please explain why CodeQL did not report an error under ubuntu, but did so under macOS?

That is a very good question. Does the `gradle` command build some C/C++ code on Linux but not on macOS? The best way to debug this would be to [re-run all jobs of the workflow with debug logging enabled](https://github.blog/changelog/2022-05-24-github-actions-re-run-jobs-with-debug-logging/). This causes the codeql-action to upload a debug artifact containing the analysed sources, logs, and other things. It should show which C/C++ files were analysed when running on Linux and the logs should show which C/C++ compiler processes were intercepted.

**Comment by aibaars** [comment]
Hmm, the error message is also a little strange `CodeQL detected code written in Java/Kotlin, Go and c, but not any written in C/C++.` . It says CodeQL detected `c` but not any `C/C++`. That doesn't make any sense. 😕

**Comment by xwm1992** [comment]
> Hmm, the error message is also a little strange `CodeQL detected code written in Java/Kotlin, Go and c, but not any written in C/C++.` . It says CodeQL detected `c` but not any `C/C++`. That doesn't make any sense. 😕

Yes, this point makes me very confused, especially this question reported under macOS. Is there any difference between `c` and `C/C++` under macOS for codeql?
